internal/server/evaluation/legacy_evaluator_test.go

```diff
@@ -2309,6 +2309,71 @@ func TestEvaluator_MatchAny_RolloutDistribution_MultiRule(t *testing.T) {
 	assert.Equal(t, flipt.EvaluationReason_MATCH_EVALUATION_REASON, resp.Reason)
 }
 
+func TestEvaluator_MatchEntityId(t *testing.T) {
+	var (
+		store  = &evaluationStoreMock{}
+		logger = zaptest.NewLogger(t)
+		s      = NewEvaluator(logger, store)
+	)
+
+	store.On("GetEvaluationRules", mock.Anything, storage.NewResource("", "foo")).Return(
+		[]*storage.EvaluationRule{
+			{
+				ID:      "1",
+				FlagKey: "foo",
+				Rank:    0,
+				Segments: map[string]*storage.EvaluationSegment{
+					"subscribers": {
+						SegmentKey: "subscribers",
+						MatchType:  flipt.MatchType_ANY_MATCH_TYPE,
+						Constraints: []storage.EvaluationConstraint{
+							{
+								ID:       "2",
+								Type:     flipt.ComparisonType_ENTITY_ID_COMPARISON_TYPE,
+								Property: "entity",
```

Comment:
could we potentially make the `property` field non-required for `type: entityID` only? i haven't looked to see if this is possible yet but just throwing it out there

Reply:
On the API level, i think we can, but not on the db level as its `NOT NULL`. We could alter the table though